cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemotePluginFinder.java

```diff
@@ -82,9 +72,9 @@
   private final RetryStrategy retryStrategy;
 
   @Inject
-  RemotePluginFinder(CConfiguration cConf, DiscoveryServiceClient discoveryServiceClient,
-                     AuthenticationContext authenticationContext,
-                     LocationFactory locationFactory) {
+  protected RemotePluginFinder(CConfiguration cConf, DiscoveryServiceClient discoveryServiceClient,
```

Comment:
An injected constructor with sub-classing is not a very good design. What's the goal of the inheritance?

Reply:
The InMemoryConfigurator relies on RemotePluginFinder, in the worker pod we need the RemotePluginFinder implementation to use the sidecar to fetch the artifact location (instead of calling app-fabric directly). 

The goal of the inheritance is to keep all the other `RemotePluginFinder` methods the same except for the `getLocation` method which will use the sidecar. The other approach I considered was creating a whole separate PluginFinder implementation but that would have a lot of duplicated code from this class.

Reply:
Spoke with Terence offline, we discussed alternatives but agreed this solution is adequate for now.